src/core/mapping/instance_manager.cc

```diff
@@ -241,8 +260,21 @@ bool InstanceSet::erase(PhysicalInstance inst)
       it++;
   }
 
-  for (auto& group : filtered_groups)
-    for (auto& region : group->regions) groups_.erase(region);
+  std::vector<Region> filtered_regions;
```

Comment:
I think we want to use a `std::set` to keep the filtered regions, as some filtered groups might have an intersection. (I believe calling `erase` with a non-existent key is safe but not efficient.)

Reply:
2c0cfb7c26a73c386d777083b46f779679b21a9a